src/ripple/app/tx/impl/details/NFTokenUtils.cpp

```diff
@@ -636,5 +674,122 @@ deleteTokenOffer(ApplyView& view, std::shared_ptr<SLE> const& offer)
     return true;
 }
 
+bool
+repairNFTokenDirectoryLinks(ApplyView& view, AccountID const& owner)
+{
+    bool didRepair = false;
+
+    auto const last = keylet::nftpage_max(owner);
+
+    std::shared_ptr<SLE> page = view.peek(Keylet(
+        ltNFTOKEN_PAGE,
+        view.succ(keylet::nftpage_min(owner).key, last.key.next())
+            .value_or(last.key)));
+
+    if (!page)
+        return didRepair;
+
+    if (page->key() == last.key)
+    {
+        // There's only one page in this entire directory.  There should be
+        // no links on that page.
+        bool const nextPresent = page->isFieldPresent(sfNextPageMin);
+        bool const prevPresent = page->isFieldPresent(sfPreviousPageMin);
+        if (nextPresent || prevPresent)
+        {
+            didRepair = true;
+            if (prevPresent)
+                page->makeFieldAbsent(sfPreviousPageMin);
+            if (nextPresent)
+                page->makeFieldAbsent(sfNextPageMin);
+            view.update(page);
+        }
+        return didRepair;
+    }
+
+    // First page is not the same as last page.  The first page should not
+    // contain a previous link.
+    if (page->isFieldPresent(sfPreviousPageMin))
+    {
+        didRepair = true;
+        page->makeFieldAbsent(sfPreviousPageMin);
+        view.update(page);
+    }
+
+    std::shared_ptr<SLE> nextPage;
+    while (
+        (nextPage = view.peek(Keylet(
+             ltNFTOKEN_PAGE,
+             view.succ(page->key().next(), last.key.next())
+                 .value_or(last.key)))))
+    {
+        if (!page->isFieldPresent(sfNextPageMin) ||
+            page->getFieldH256(sfNextPageMin) != nextPage->key())
+        {
+            didRepair = true;
+            page->setFieldH256(sfNextPageMin, nextPage->key());
+            view.update(page);
```

Comment:
From my understanding, this code checks if `sfNextPageMin` of the current page links correctly to the next page.

But is this related to the address corruption bug? It seems to me this has nothing to do with the bug...

Reply:
Im kind of curious if this code is here for safety, or is it because some page is actually mislinked?

Reply:
I put in the code that looks for missing links because that's what we actually see in the ledger.  I added the code that verifies the link is valid because I was here and the check was cheap to do.

I've seen no instances of pages that were linked with incorrect values in the wild.  I've only seen links that are missing.